controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller_test.go

```diff
@@ -0,0 +1,233 @@
+package spaceprovisionerconfig
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/apis"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	. "github.com/codeready-toolchain/toolchain-common/pkg/test/assertions"
+	. "github.com/codeready-toolchain/toolchain-common/pkg/test/spaceprovisionerconfig"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestSpaceProvisionerConfigValidation(t *testing.T) {
+	t.Run("is not valid when non-existing ToolchainCluster is referenced", func(t *testing.T) {
+		// given
+		spc := NewSpaceProvisionerConfig("spc", test.HostOperatorNs, ReferencingToolchainCluster("non-existent"))
+		r, req, cl := prepareReconcile(t, spc)
+
+		// when
+		_, reconcileErr := r.Reconcile(context.TODO(), req)
+		require.NoError(t, cl.Get(context.TODO(), runtimeclient.ObjectKeyFromObject(spc), spc))
+
+		// then
+		assert.NoError(t, reconcileErr)
+		AssertThat(t, spc, Is(NotReadyWithReason(toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason)))
```

Comment:
related to my comment in the other PR:
https://github.com/codeready-toolchain/toolchain-common/pull/357#discussion_r1471088952
I just tried what would happen if I changed the test to
```
AssertThat(t, spc, Is(Ready()))
```
I know that it's wrong, but I wanted to know what the error would look like, where it actually fails and how easy it is to debug. The error looks like this:
```
--- FAIL: TestSpaceProvisionerConfigValidation (0.02s)
    --- FAIL: TestSpaceProvisionerConfigValidation/is_not_valid_when_non-existing_ToolchainCluster_is_referenced (0.01s)
        assertions.go:27: 
            	Error Trace:	/home/mjobanek/go-workspace/pkg/mod/github.com/metlos/toolchain-common@v0.0.0-20240125174611-654ac7c7e9a7/pkg/test/assertions/assertions.go:27
            	            				/home/mjobanek/go-workspace/src/github.com/codeready-toolchain/host-operator/controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller_test.go:36
            	Error:      	Should be true
            	Test:       	TestSpaceProvisionerConfigValidation/is_not_valid_when_non-existing_ToolchainCluster_is_referenced
```
where the `assertions.go:27` line points to:
```
func AssertThat(t *testing.T, object client.Object, predicate Predicate[client.Object]) {
	assert.True(t, predicate.Matches(object))
}
```
so it's very hard to understand what was expected and what was the actual value.
TBH, it's fine in this particular case since there is only one expectation that wasn't met, but we should be very careful as soon as we have multiple assertions.

Reply:
Excellent point. I think the solution here is to mark `AssertThat` as a helper method so that it is skipped when reporting the location of the error. Here's the PR: https://github.com/codeready-toolchain/toolchain-common/pull/360